app/transformations/MemberPaymentsTransformer.scala

```diff
@@ -317,6 +377,74 @@ class MemberPaymentsTransformer @Inject()(
+   *   - if there are previous user answers, compare members:
+   *    - if they are the same, they are not new members.
+   *    - if they are different, check if they were added in the latest version of the return (memberStatus is NEW and return’s fbVersion == memberPSRVersion)
+   *      - if so, they are changed members.
+   *      - if not, they are new members.
+   *
+   * fetchingPreviousVersion - used to indicate that the PSR retrieval service is calling this method when fetching the previous version
+   *                         - this is required as when this happens, there won't be a previousVersionUA yet
+   *                         - we can't rely on the absence of previousVersionUA to determine this as the PSR connector can return None if not found and in some cases it is required to be there
+   */
+  private def identifyNewMembers(
+    srn: Srn,
+    ua: UserAnswers,
+    previousVersionUA: Option[UserAnswers],
+    fetchingPreviousVersion: Boolean
+  ): Try[List[Max300]] =
+    buildMemberDetails(srn, ua).left.map(new Exception(_)).toTry.flatMap { currentMemberDetails =>
+      previousVersionUA match {
+        case Some(_) if fetchingPreviousVersion =>
+          Failure(
+            new Exception(
+              "fetchingPreviousVersion flag is true but previous PSR version already exists. Something went wrong"
+            )
+          )
+        case Some(previousUA) =>
+          logger.info(s"[identifyNewMembers] Previous PSR version found for srn $srn")
+          buildMemberDetails(srn, previousUA).left.map(new Exception(_)).toTry.flatMap { previousMemberDetails =>
+            Success(
+              currentMemberDetails.toList.flatMap {
+                case (index, currentMemberDetail) if previousMemberDetails.get(index).contains(currentMemberDetail) =>
+                  None
+                case (_, currentMemberDetail) if currentMemberDetail.state.changed => None
+                case (index, currentMemberDetail) if currentMemberDetail.state._new => Some(index)
+                case _ => None
+              }
+            )
+          }
+        // No previous version UserAnswers so this must be either pre-first submission or just after the first submission
+        case None =>
+          val (membersWithNoVersions, membersWithVersions): (List[Max300], List[(Max300, String)]) =
+            currentMemberDetails.toList.partitionMap {
+              case (index, memberDetails) if memberDetails.memberPSRVersion.isEmpty => Left(index)
+              case (index, MemberDetails(_, Some(version), _, _, _, _, _, _, _, _)) => Right(index -> version)
+            }
+
+          val versionedMembersWithETMPStatus = membersWithVersions.map {
+            case (index, memberVersion) => (index, memberVersion, ua.get(MemberStatus(srn, index)))
+          }
+
+          Success(
+            versionedMembersWithETMPStatus.flatMap {
+              // Added in current version of the submission, safe to soft delete
```

Comment:
Should this say "safe to hard delete", or am I getting mixed up?

Reply:
you're right - to not trigger another build i'll fix it in the next PR 👍